normandy/control/static/control/js/components/ActionForm.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { reduxForm } from 'redux-form'
+import { _ } from 'underscore'
+import HeartbeatForm from './action_forms/HeartbeatForm.jsx'
+import ConsoleLogForm from './action_forms/ConsoleLogForm.jsx'
+
+class ActionForm extends React.Component {
+  constructor(props) {
+    super(props)
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const currentProps = _.pick(this.props, 'initialValues', 'values', 'fields', 'selectedAction');
+    const incomingProps = _.pick(nextProps, 'initialValues', 'values', 'fields', 'selectedAction');
+
+    return (!_.isEqual(currentProps, incomingProps) || !_.isEqual(this.state, nextState));
```

Comment:
I don't understand how you picked these 4 files. It seems like you only use `fields` and `arugments_schema` in `render()`. Why include `selectedAction`, `initialValues`, and `values`? Why not include `argument_schema`?

Was there a performance problem without this? I can't really imagine calling render to really be that expensive if the DOM doesn't change.